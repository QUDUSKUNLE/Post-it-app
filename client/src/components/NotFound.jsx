import React from 'react';
import { Link } from 'react-router-dom';

/**
 * displays an error page when user navigates to an alien page
 * @function NotFound
 * @return {void} - React component
 */
const NotFound = () => (
  <div className="col-md-12">
    <div className="inner-content center m-auto">
      <h3 className="text-center">Page Not Found.</h3>
      <Link to="/signin">Go Back</Link>
    </div>
  </div>
);

export default NotFound;
