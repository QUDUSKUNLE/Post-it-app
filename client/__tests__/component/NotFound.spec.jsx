import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import NotFound from '../../src/components/NotFound';

describe('<NotFound/>', () => {
  it('component should be defined', () => {
    expect(NotFound).toBeDefined();
  });
  it('should render correctly', () => {
    const component = shallow(<NotFound />);
    expect(component).toMatchSnapshot();
    expect(component.find('.notFound')).toHaveLength(1);
    expect(component.find('div')).toHaveLength(3);
  });
});
