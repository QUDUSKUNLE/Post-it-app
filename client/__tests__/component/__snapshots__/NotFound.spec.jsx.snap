// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound/> should render correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="col-md-12"
  >
    <div
      className="inner-content center m-auto"
    >
      <h3
        className="text-center"
      >
        Page Not Found.
      </h3>
      <Link
        replace={false}
        to="/signin"
      >
        Go Back
      </Link>
    </div>
  </div>,
  "nodes": Array [
    <div
      className="col-md-12"
    >
      <div
        className="inner-content center m-auto"
      >
        <h3
          className="text-center"
        >
          Page Not Found.
        </h3>
        <Link
          replace={false}
          to="/signin"
        >
          Go Back
        </Link>
      </div>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <NotFound />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="col-md-12"
        >
          <div
            className="inner-content center m-auto"
          >
            <h3
              className="text-center"
            >
              Page Not Found.
            </h3>
            <Link
              replace={false}
              to="/signin"
            >
              Go Back
            </Link>
          </div>
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="col-md-12"
        >
          <div
            className="inner-content center m-auto"
          >
            <h3
              className="text-center"
            >
              Page Not Found.
            </h3>
            <Link
              replace={false}
              to="/signin"
            >
              Go Back
            </Link>
          </div>
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <NotFound />,
}
`;
